### PR TITLE
Add keyboard hifumi

### PR DESCRIPTION
**Pull request by zk-phi**
## Description

Add configurations for a new keyboard, hifumi.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None.

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hifumi/Readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# hifumi
+
+A macro pad with 6 keys.
+
+Keyboard Maintainer: riconken, [zk-phi](https://github.com/zk-phi)
```
Comment (mechmerlin) [inline keyboards/hifumi/Readme.md]: You need to add some spaces here for markdown to render properly

```suggestion
Keyboard Maintainer: riconken, [zk-phi](https://github.com/zk-phi)  
```

File: keyboards/hifumi/Readme.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Hardware Supported: hifumi PCB, Pro Micro
```
Comment (mechmerlin) [inline keyboards/hifumi/Readme.md]:
```suggestion
Hardware Supported: hifumi PCB, Pro Micro   
```

File: keyboards/hifumi/Readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Hardware Availability: [PCB & Case Data](https://github.com/zk-phi/1x2x3-keyboard)
```
Comment (mechmerlin) [inline keyboards/hifumi/Readme.md]:
```suggestion
Hardware Availability: [PCB & Case Data](https://github.com/zk-phi/1x2x3-keyboard)  
```

File: keyboards/hifumi/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "hifumi.h"
```
Comment (mechmerlin) [inline keyboards/hifumi/keymaps/default/keymap.c]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/hifumi/keymaps/test/keymap.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "hifumi.h"
```
Comment (mechmerlin) [inline keyboards/hifumi/keymaps/test/keymap.c]:
```suggestion
#include QMK_KEYBOARD_H
```

**Comment by zk-phi** [comment]
Thank you for reviewing, and suggestions!

File: keyboards/hifumi/rules.mk [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+BOOTLOADER = caterina
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/hifumi/rules.mk]: Does this keyboard have a physical reset button?  

If not, you may want to enable Bootmagic (and probably, bootmagic light), so there is a way to jump into the bootloader, easily.

```suggestion
BOOTMAGIC_ENABLE = lite       # Virtual DIP switch configuration(+1000)
```

File: keyboards/hifumi/Readme.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# hifumi
```
Comment (noroadsleft) [inline keyboards/hifumi/Readme.md]: This file should be named as `readme.md`, fully lowercase.

File: keyboards/hifumi/Readme.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+Keyboard Maintainer: riconken, [zk-phi](https://github.com/zk-phi)  
+Hardware Supported: hifumi PCB, Pro Micro   
+Hardware Availability: [PCB & Case Data](https://github.com/zk-phi/1x2x3-keyboard)  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hifumi:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/hifumi/Readme.md]: Could you please change this as follows?

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/hifumi/config.h [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 3
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
```
Comment (noroadsleft) [inline keyboards/hifumi/config.h]: You don't have a `DIODE_DIRECTION` defined. See Lines 39-40 for example.

https://docs.qmk.fm/#/config_options?id=hardware-options

File: keyboards/hifumi/rules.mk (hunk 6 above)
Comment (zk-phi) [reply keyboards/hifumi/rules.mk]: Yes this keyboard has a reset button!

**Comment by zk-phi** [comment]
Thankyou drashna and noroadsleft too! Fixed.

File: keyboards/hifumi/rules.mk (hunk 6 above)
Comment (drashna) [reply keyboards/hifumi/rules.mk]: Okay, then this isn't needed!

I just wanted to make sure.

File: keyboards/hifumi/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        _______, KC_PGUP, MO(ADJUST),
+        KC_HOME, KC_PGDN, KC_END
+    ),
+    [LOWER] = LAYOUT(
+        MO(ADJUST), _______, _______,
+        _______,    _______, _______
+    ),
+    [ADJUST] = LAYOUT(
+        _______,        RGB_TOG,        _______,
+        RGB_MODE_SNAKE, RGB_MODE_PLAIN, RGB_HUI
```
Comment (drashna) [inline keyboards/hifumi/keymaps/default/keymap.c]: There are short forms of these keycodes:
https://docs.qmk.fm/#/feature_rgblight?id=keycodes

```suggestion
        RGB_M_SN, RGB_M_P, RGB_HUI
```

**Comment by drashna** [comment]
It looks like this board may be more suited for the `handwired` folder, as it's not a mass produced (or even in a group buy?).

**Comment by drashna** [comment]
It looks like this should be in the 'handwired` folder. 

Since it looks like this isn't a mass produced board, we try to limit boards to that folder (though, we don't always catch that/strictly enforce that).

**Comment by zk-phi** [comment]
Yes the keyboard has not been sold in mass yet, but we're selling the kit (PCB and acrylic) first in a tech event, held at 14th Apr in Japan :)
Then we'd like to make the firmware available at `keyboards/hifumi` in advance, if it's accepted.

**Comment by drashna** [comment]
Ah, okay!  In that case, keep it out of `handwired` then. 

And good luck with the kits! 

Also, in case you were not aware, once the code for the board is merged, a full firmware image will be available at `https://qmk.fm/keyboards` and will show up in the [QMK Configurator](https://config.qmk.fm/).   Both of which can be easily linked.

**Comment by zk-phi** [comment]
Thank you!
I'll try it out too > QMK Configurator

**Comment by drashna** [comment]
Well, it's pushed now (it's done automatically). 

And you can find it here: https://config.qmk.fm/#/hifumi/LAYOUT

**Comment by zk-phi** [comment]
Cool, thanks (also shared with the members) !

**Comment by drashna** [comment]
You're very welcome! 
And glad to hear it!

And again, good luck with the kits!

**Pull request merged**